tests/python/test_serial_execution.py

```diff
@@ -0,0 +1,37 @@
+import taichi as ti
+
+
+@ti.test(arch=ti.cpu, cpu_max_num_threads=1)
+def test_serial_range_for():
+    n = 1024 * 32
+    s = ti.field(dtype=ti.i32, shape=n)
+    counter = ti.field(dtype=ti.i32, shape=())
+
+    @ti.kernel
+    def fill_range():
+        counter[None] = 0
+        for i in range(n):
+            s[ti.atomic_add(counter[None], 1)] = i
+
+    fill_range()
+
+    for i in range(n):
+        assert s[i] == i
```

Comment:
Why not use s.to_numpy and np.arange? This would be slow.

Reply:
It actually finishes < 1s. I'd prefer a simpler solution here.